dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc

```diff
@@ -180,4 +181,63 @@ TYPED_TEST(nvjpegDecodeDecoupledAPITest, TestSingleTiffDecode4T) {
   this->TiffTestDecode(4);
 }
 
+class HwDecoderUtilizationTest : public ::testing::Test {
+ public:
+  void SetUp() final {
+    dali::string list_root(testing::dali_extra_path() + "/db/single/jpeg");
+
+    pipeline_.AddOperator(
+            OpSpec("FileReader")
+                    .AddArg("device", "cpu")
+                    .AddArg("file_root", list_root)
+                    .AddOutput("compressed_images", "cpu")
+                    .AddOutput("labels", "cpu"));
+    auto decoder_spec =
+            OpSpec("ImageDecoder")
+                    .AddArg("device", "mixed")
+                    .AddArg("output_type", DALI_RGB)
+                    .AddArg("hw_decoder_load", .7f)
+                    .AddInput("compressed_images", "cpu")
+                    .AddOutput("images", "gpu");
+    pipeline_.AddOperator(decoder_spec, decoder_name_);
+
+    pipeline_.Build(outputs_);
+
+    auto node = pipeline_.GetOperatorNode(decoder_name_);
+    if (!node->op->GetDiagnostic<bool>("using_hw_decoder")) {
+      try {
+        if (nvml::HasHwDecoder()) {
+          FAIL() << "HW Decoder exists in the system and failed to open";
+        }
+      } catch (const std::runtime_error &e) {
+        DALI_WARN(
+                make_string("NVML API is unsupported. Please update the CUDA driver. ", e.what()));
```

Comment:
shouldn't this be an error?

Reply:
If the `HasHwDecoder` function throws, it means, that NVML is outdated. This can be legit, if we have old driver and CUDA10 build, so in this case test should skip, not fail. Hence the warning.